app/Material.php

```diff
@@ -47,4 +49,41 @@ public function isLost()
     {
         return !is_null($this->returner_registration_mark);
     }
+
+    /**
+     * Scope a query to only return borrowable materials
+     *
+     * @param Builder $query
+     * @return Builder $query
+     */
+    public function scopeBorrowable($query)
+    {
+        return $query->where([
+            ['amount', '>=', 1],
+            ['returner_registration_mark', '=', null],
+            ['tooker_registration_mark', '=', null]
+        ]);
+    }
+
+    public function hasBorrowableWithName(string $name)
+    {
+        $material = $this->borrowable()->where('name', $name)->first();
+        return isset($material);
+    }
```

Comment:
Transform that function to a static function and call the borrowable with ```self```, so that it no need to be instanced.

Reply:
I will use it in a instance object any way. It's unneccessary. I would have to do `$material = (new self)->borrowable()->where('name', $name)->first();`. And the application seems almost finished. If another function need that method I will turn it into static, but for now doesn't make sense.

Reply:
No you will do ```self::borrowable()```. In the documentation the local scopes could be call without a instance.
![image](https://user-images.githubusercontent.com/44510574/84537954-20623d80-acc7-11ea-8f8d-2a026ad0f257.png)

Reply:
self == class
$this == instance